src/researchhub_comment/related_models/rh_comment_model.py

```diff
@@ -77,29 +78,39 @@ def is_root_comment(self):
     """ --- METHODS --- """
 
     @classmethod
-    def create_from_request(cls, request, rh_thread):
-        request_data = request.data
-        [
-            comment_content_src_file,
-            comment_content_type,
-        ] = cls.get_comment_src_file_from_request(request)
-        rh_comment = cls.objects.create(
-            thread=rh_thread,
-            parent=request_data.get("comment_parent_id"),
-            comment_content_type=comment_content_type,
+    def create_from_data(cls, data, rh_thread):
+        from researchhub_comment.serializers.rh_comment_serializer import (
+            RhCommentSerializer,
         )
-        rh_comment.comment_content_src.save(
-            f"RH-THREAD-{rh_thread.id}-COMMENT-{rh_comment.id}-user-{request.user.id}.txt",
-            comment_content_src_file,
-        )
-        rh_comment.refresh_from_db()
-        return rh_comment
+        with transaction.atomic():
```

Comment:
We can get rid of one of these atomic transactions. Either the one in the view or the mixin.